### PR TITLE
main : check --config-check should has --config flag specified first.

### PR DESCRIPTION
**Pull request by AilinKid**
### What problem does this PR solve? 
```
when start tidb with flag: 
 --config-check & **without** --config specified
it will succeed in config check regardless of what content the config file has. 
```

### What is changed and how it works?

Make sure --config is specified first when it has --config-check flag, then talk about file validation.
Otherwise, return config check failed cause no config file specified.

### Check List 

Tests 

manual test

Code changes

modified in loadconfig.go

Side effects

no

Related changes

no

Release Note

check  --config-check should have --config flag specified first.

### CODE REVIEW

File: tidb-server/main.go [hunk 1]
```diff
@@ -345,6 +345,12 @@ func loadConfig() string {
 			return err.Error()
 		}
 		terror.MustNil(err)
+	} else {
+		// configCheck should have the config file specified
+		if *configCheck {
+			fmt.Fprintln(os.Stderr, "config check failed", errors.New("no config file specified for config-check"))
+			os.Exit(1)
```
Comment (zimulala) [inline tidb-server/main.go]: Could we use `MustNil`?

File: tidb-server/main.go [hunk 2]
```diff
@@ -345,6 +345,12 @@ func loadConfig() string {
 			return err.Error()
 		}
 		terror.MustNil(err)
+	} else {
+		// configCheck should have the config file specified
```
Comment (zimulala) [inline tidb-server/main.go]: Please add `.` at the end of a line.

File: tidb-server/main.go (hunk 1 above)
Comment (AilinKid) [reply tidb-server/main.go]: Em... it's not in test file .
This is same as previous config valid code when tidb start, `Fprintln` to Stderr with err. 
Cause user specified no config file, we new an error here.

File: tidb-server/main.go (hunk 2 above)
Comment (AilinKid) [reply tidb-server/main.go]: Ok, Done

File: tidb-server/main.go (hunk 1 above)
Comment (zimulala) [reply tidb-server/main.go]: I know it isn't the test file, you can see `MustNil` is called on line 347.

Comment (AilinKid) [reply tidb-server/main.go]: Enen, got it. I've update my comment

**Comment by Deardrops** [comment]
LGTM

**Comment by Deardrops** [comment]
LGTM

**Pull request merged**